cli/tests/common/gitoxide.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2025 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::Path;
+use std::path::PathBuf;
+
+pub const GIT_USER: &str = "Someone";
+pub const GIT_EMAIL: &str = "someone@example.org";
+
+#[allow(unused)]
+pub fn open(directory: impl Into<PathBuf>) -> gix::Repository {
+    let repo = gix::open(directory).unwrap();
+    set_config(repo)
+}
+
+#[allow(unused)]
+pub fn init(directory: impl AsRef<Path>) -> gix::Repository {
+    let repo = gix::init(directory).unwrap();
+    set_config(repo)
+}
+
+#[allow(unused)]
+pub fn init_bare(directory: impl AsRef<Path>) -> gix::Repository {
+    let repo = gix::init_bare(directory).unwrap();
+    set_config(repo)
+}
+
+// gitoxide looks at the user's git config
+// resetting it prevents the tests from relying on the user config
+// moreover, there is a bug in gitoxide which means that `commit_as` is broken
+#[allow(unused)]
+pub fn set_config(mut repo: gix::Repository) -> gix::Repository {
+    let mut config_mut = repo.config_snapshot_mut();
+    let mut new_config = gix::config::File::default();
+    let mut user_section = new_config.new_section("user", None).unwrap();
+    user_section.set("name".try_into().unwrap(), GIT_USER.into());
+    user_section.set("email".try_into().unwrap(), GIT_EMAIL.into());
+    *config_mut = new_config;
+    drop(config_mut);
+    repo
+}
```

Comment:
nit: maybe we can use `gix::open::Options::default().config_overrides()` instead? See git_backend.rs.